frontend/packages/ux-editor/src/components/config/Expressions/DataSourceValue.tsx

```diff
@@ -82,15 +83,16 @@ export const DataSourceValue = ({
       );
     case DataSource.String:
       return (
-        <LegacyTextField
+        <Textfield
           onChange={(e) => specifyDataSourceValue(e.target.value, isComparableValue)}
           value={currentValue as string}
         />
       );
     case DataSource.Number:
       return (
-        <LegacyTextField
-          formatting={{ number: {} }}
+        <Textfield
+          data-testid={testids.textboxNumber}
+          type='number'
```

Comment:
We should avoid using type="number" on input-fields which should contain a number[. Ref. web-docs type="number" should only be used if the input-field is for incremental numbers.](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/number#using_number_inputs).

We can use the inputMode='numeric' to get a numeric keyboard for touch-devices. I suggest that we revert this to the old component, until we have a number-component with validation.

Reply:
Reverted 👍